### ISSUE TITLE
Parameters from the `Accept` request HTTP header are ignored during content negotiation

### ISSUE THREAD

**Issue by kamilkloch** [softwaremill/tapir#3232]
As per https://www.rfc-editor.org/rfc/rfc9110.html#name-content-negotiation-fields, `Accept` header may contain additional parameters, i.e.:
```
Accept: application/json;IEEE754Compatible=true
```
Those parameters are ignored by tapir during content negotiation. Below we define an endpoint with distinct codecs for 2 media types:
- `application/json`,
- `application/json;IEEE754Compatible=true`
.
```scala
  def circeCodec1[T: Encoder : Decoder : Schema]: JsonCodec[T] = {
    val c = circeCodec[T]
    sttp.tapir.Codec.anyString(CodecFormat.Json())(c.decode)(c.encode)
  }

  def circeCodec2[T: Encoder : Decoder : Schema]: Codec[String, T, JsonIEEE754CompatibleCodecFormat] = {
    val c = circeCodec[T]
    sttp.tapir.Codec.anyString(JsonIEEE754CompatibleCodecFormat())(c.decode)(c.encode(_).pipe(v => s"CUSTOMIZED2: $v"))
  }

  case class JsonIEEE754CompatibleCodecFormat() extends CodecFormat {
    override val mediaType: MediaType = MediaType.ApplicationJson.copy(otherParameters = Map("IEEE754Compatible" -> "true"))
  }

  def jsonBody1[T: Encoder : Decoder : Schema]: EndpointIO.Body[String, T] = {
    val c: Codec[String, T, CodecFormat.Json] = circeCodec1[T]
    EndpointIO.Body(RawBodyType.StringBody(StandardCharsets.UTF_8), c, EndpointIO.Info.empty)
  }

  def jsonBody2[T: Encoder : Decoder : Schema]: EndpointIO.Body[String, T] = {
    val c: Codec[String, T, JsonIEEE754CompatibleCodecFormat] = circeCodec2[T]
    EndpointIO.Body(RawBodyType.StringBody(StandardCharsets.UTF_8), c, EndpointIO.Info.empty)
  }

  case class Payload(x: Double)

  val contentEndpoint = endpoint.get
    .in("ts")
    .out(oneOfBody(
      jsonBody1[Payload],
      jsonBody2[Payload],
    ))
```

Curl results:
```
curl -v --header "Accept: application/json" localhost:8888/ts
*   Trying 127.0.0.1:8888...
* Connected to localhost (127.0.0.1) port 8888 (#0)
> GET /ts HTTP/1.1
> Host: localhost:8888
> User-Agent: curl/7.88.1
> Accept: application/json
> 
< HTTP/1.1 200 OK
< Content-Type: application/json
< Date: Mon, 09 Oct 2023 14:25:50 GMT
< Content-Length: 9
< 
* Connection #0 to host localhost left intact
{"x":1.0}
```
```
curl -v --header "Accept: application/json;IEEE754Compatible=true" localhost:8888/ts
*   Trying 127.0.0.1:8888...
* Connected to localhost (127.0.0.1) port 8888 (#0)
> GET /ts HTTP/1.1
> Host: localhost:8888
> User-Agent: curl/7.88.1
> Accept: application/json;IEEE754Compatible=true
> 
< HTTP/1.1 200 OK
< Content-Type: application/json
< Date: Mon, 09 Oct 2023 14:26:04 GMT
< Content-Length: 9
< 
* Connection #0 to host localhost left intact
{"x":1.0}
```

In both cases `jsonBody1` codec is applied, which is arguably an incorrect behavior.

**Comment by adamw** [comment]
Indeed, seems we should take into account the parameters during content negotiation

**Comment by kamilkloch** [comment]
Will adding `otherParameters: Map[String, String]` to `ContentTypeRange` suffice?

**Comment by kamilkloch** [comment]
Fixed in #3243.
